hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -331,3 +337,119 @@ func (a OpenStack) DeleteCredentials(ctx context.Context, c client.Client, hclus
 	// }
 	return nil
 }
+
+func machineNetworksToStrings(machineNetworks []hyperv1.MachineNetworkEntry) []string {
+	var machineNetworksStr []string
+	for _, machineNetwork := range machineNetworks {
+		machineNetworksStr = append(machineNetworksStr, machineNetwork.CIDR.String())
+	}
+	return machineNetworksStr
+}
+
+func defaultWorkerSecurityGroupRules(machineCIDRs []string) []capo.SecurityGroupRuleSpec {
+	ingressRules := []capo.SecurityGroupRuleSpec{}
+
+	// Rules for worker security group with the remote IP prefix set to the machine CIDRs
+	for _, machineCIDR := range machineCIDRs {
+		machineCIDRInboundRules := []capo.SecurityGroupRuleSpec{
+			{
+				Name:     "esp-ingress",
+				Protocol: k8sutilspointer.String("esp"),
+			},
+			{
+				Name:     "icmp-ingress",
+				Protocol: k8sutilspointer.String("icmp"),
+			},
+			{
+				Name:         "router-ingress",
+				Protocol:     k8sutilspointer.String("tcp"),
+				PortRangeMin: k8sutilspointer.Int(1936),
+				PortRangeMax: k8sutilspointer.Int(1936),
+			},
+			{
+				Name:         "ssh-ingress",
+				Protocol:     k8sutilspointer.String("tcp"),
+				PortRangeMin: k8sutilspointer.Int(22),
+				PortRangeMax: k8sutilspointer.Int(22),
+			},
+			{
+				Name:     "vrrp-ingress",
+				Protocol: k8sutilspointer.String("vrrp"),
+			},
+		}
+
+		for i, rule := range machineCIDRInboundRules {
+			rule.RemoteIPPrefix = k8sutilspointer.String(machineCIDR)
+			machineCIDRInboundRules[i] = rule
+		}
+
+		ingressRules = append(ingressRules, machineCIDRInboundRules...)
```

Comment:
the rule for port 10250, which is for kubelet insecure and 30000-32767 for ingress services are already created by default by CAPO  right?

Reply:
* 10250: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/20e8bac7adbbced3949d92c8d34850b35a28dd4a/pkg/cloud/services/networking/securitygroups_rules.go#L76-L98
* 30000-32767: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/20e8bac7adbbced3949d92c8d34850b35a28dd4a/pkg/cloud/services/networking/securitygroups_rules.go#L145-L164